lib/msal-common/src/cache/entities/AppMetadataEntity.ts

```diff
@@ -27,7 +27,19 @@ export class AppMetadataEntity {
      * Generate Account Cache Key as per the schema: <home_account_id>-<environment>-<realm*>
      */
     generateAppMetaDataEntityKey(): string {
-        const appMetaDataKeyArray: Array<string> = [APP_META_DATA, this.environment, this.clientId];
+        const appMetaDataKeyArray: Array<string> = [APP_METADATA, this.environment, this.clientId];
         return appMetaDataKeyArray.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
     }
+
+    /**
+     * Validates an entity: checks for all expected params
+     * @param entity
+     */
+    static isAppMetadataEntity(key: string, entity: object): boolean {
+        return (
+            (key.indexOf(APP_METADATA) !== -1) &&
```

Comment:
If the key is expected to start with "appmetadata" you can check if `key.indexOf(APP_METADATA) === 0` here

Reply:
yes. Good call.